packages/flutter_hooks/lib/src/platform_brightness.dart

```diff
@@ -0,0 +1,60 @@
+part of 'hooks.dart';
+
+/// A callback triggered when the platform brightness changes.
+typedef BrightnessCallback = FutureOr<void> Function(
+  Brightness? brightness,
+);
+
+/// Returns the current Platform Brightness value and rebuilds the widget when it changes.
+Brightness? usePlatformBrightness() {
+  return use(const _PlatformBrightnessHook(rebuildOnChange: true));
+}
+
+useOnPlatformBrightnessChange(BrightnessCallback? onBrightnessChange) {
+  return use(_PlatformBrightnessHook(onBrightnessChange: onBrightnessChange));
+}
+
+class _PlatformBrightnessHook extends Hook<Brightness?> {
+  const _PlatformBrightnessHook({
+    this.rebuildOnChange = false,
+    this.onBrightnessChange,
+  }) : super();
+
+  final bool rebuildOnChange;
+  final BrightnessCallback? onBrightnessChange;
+
+  @override
+  _PlatformBrightnessState createState() => _PlatformBrightnessState();
+}
+
+class _PlatformBrightnessState
+    extends HookState<Brightness?, _PlatformBrightnessHook>
```

Comment:
Needs to override debugLabel/debugValue  
See https://github.com/rrousselGit/flutter_hooks/blob/c87b6d947b3b5cb1b1b210f121ce752b767da748/packages/flutter_hooks/lib/src/async.dart#L114